src/interpreter.rs

```diff
@@ -9,6 +10,7 @@ pub enum Value {
     Str(String),
     Num(f64),
     Arr(),
+    Pipeline(process::Output),
```

Comment:
Don't you also need stderr and perhaps the exit code here?  I don't see how stderr is handled, which is critical for knowing if/why your command failed.

Reply:
process::Output is a struct containing, stdout, stderr and Exit Status.

https://doc.rust-lang.org/std/process/struct.Output.html